doc/configuration.rst

```diff
@@ -499,7 +499,7 @@ You need to set the ``request_format`` to ``image/png`` when using ``mixed``-mod
 ``request_format``
 """"""""""""""""""
 
-MapProxy will try to use this format to request new tiles, if it is not set ``format`` is used. This option has no effect if the source does not support that format or the format of the source is set explicitly (see ``suported_format`` or ``format`` for sources).
+MapProxy will try to use this format to request new tiles, if it is not set ``format`` is used. This option has no effect if the source does not support that format or the format of the source is set explicitly (see ``format`` for sources).
```

Comment:
`supported_formats` exists for sources, `format` exists for caches.

https://mapproxy.github.io/mapproxy/latest/sources.html#supported-formats
https://mapproxy.github.io/mapproxy/latest/configuration.html#format


```suggestion
MapProxy will try to use this format to request new tiles, if it is not set ``format`` is used. This option has no effect if the source does not support that format or the format of the cache is set explicitly (see ``format`` for caches or ``supported_formats`` for sources).
```
We have to check the code if this actually is the case here.

Reply:
Good catch! `supported_formats` seems to work indeed. I used this `mapproxy.yaml` to test it:

```
# start via: mapproxy-util serve-develop mapproxy.yaml -b :1233
# example query: http://localhost:1233/tms/1.0.0/italy_parcels/webmercator/18/280207/333564.png
services:
  tms:
    use_grid_names: true
    origin: 'nw'
  demo:

layers:
  - name: italy_parcels
    title: ItalyParcels
    sources: [ItalyCache]
    
caches:
  ItalyCache:
    grids: [webmercator]
    sources: [ItalyWMS]
    cache:
      type: file
      directory: cached_tms
      directory_layout: tms

sources:
  ItalyWMS:
    type: wms
    req:
      url: https://wms.cartografia.agenziaentrate.gov.it/inspire/wms/ows01.php?language=ita&
      layers: CP.CadastralParcel,strade,vestizioni,province,codice_plla,fabbricati,CP.CadastralZoning,acque,simbolo_graffa,copyright
    supported_srs: ['EPSG:25832']
    supported_formats: ['image/png'] # or ['image/jpeg']

grids:
  webmercator:
    base: GLOBAL_WEBMERCATOR

globals:
```

Using the above file, a request which is made to the WMS server is the following:

> [2024-07-16 11:06:45,833] mapproxy.source.request - INFO - GET https://wms.cartografia.agenziaentrate.gov.it/inspire/wms/ows01.php?language=ita&&layers=CP.CadastralParcel,strade,vestizioni,province,codice_plla,fabbricati,CP.CadastralZoning,acque,simbolo_graffa,copyright&format=image%2Fpng&bbox=772823.3043114123,4872949.21939256,773088.581247699,4873213.6462514745&width=1188&height=1184&srs=EPSG%3A25832&request=GetMap&version=1.1.1&service=WMS&styles= 200 - 1008

ie data is requested in PNG. If I change `['image/png']` to `['image/jpeg']`, delete `cached_tms` and relaunch, the new request is using JPEG:

> [2024-07-16 11:09:50,055] mapproxy.source.request - INFO - GET https://wms.cartografia.agenziaentrate.gov.it/inspire/wms/ows01.php?language=ita&&layers=CP.CadastralParcel,strade,vestizioni,province,codice_plla,fabbricati,CP.CadastralZoning,acque,simbolo_graffa,copyright&format=image%2Fjpeg&bbox=772823.3043114123,4872949.21939256,773088.581247699,4873213.6462514745&width=1188&height=1184&srs=EPSG%3A25832&request=GetMap&version=1.1.1&service=WMS&styles= 200 - 1065